src/librbd/CMakeLists.txt

```diff
@@ -107,9 +107,12 @@ target_link_libraries(librbd LINK_PRIVATE
   cls_journal_client 
   common
   pthread
-  udev
   ${CMAKE_DL_LIBS}
   ${EXTRALIBS})
+if(HAVE_UDEV)
```

Comment:
> I try to change as little functionality as possible when adding FreeBSD/Clang specifics.

i agree, but behaviorwise, removing libudev here does not change anything.

> I'll make a separate PR for that, and rebase this one.

thanks.